### DISCUSSION TITLE
Jupyter Book TOC right-hand margin in the way

### DISCUSSION THREAD

**Discussion by chipbrock** [executablebooks/meta#599] [Q&A]
I really enjoy Jupyter Books and am actively involved in two book-length projects, one for print and one for on-line. I use the margin captions (all Tufte) for my figures but it's almost always overwritten by the TOC which runs through the right hand margin. I would have thought that it should be possible to show/hide the TOC in the browser, but I cannot find any documentation to set that up. Here's a typical example of my figure code:

````
<!--....figure.....................................-->
```{figure} ../_images/EM/pluck.png
---
width: 90%
figclass: margin-caption
alt: My figure text
name: myfig5
---
A "pluck" of a string transmits the potential energy of the displacement of the particles in the string through the tension and relaxation of the string. The disturbance moves away, while the bits of string stay put.
```
<!--.........................................-->
````
Attached is a screen shot showing the occlusion.  

Am I missing something?

thanks

<img width="1081" alt="modified" src="https://user-images.githubusercontent.com/4515277/147681963-14347357-dfc6-4464-8939-b7a7fddb6ab2.png">

**Comment by choldgraf** [comment]
Hmmm - in general, the right TOC should hide itself when there's margin content on the page. Do you have a link to a live site?

**Reply by chipbrock** [reply]
thanks for your quick response.

Yes, here’s a link to an actually-served page:

https://qstbb.pa.msu.edu/storage/ISP220_fall2020/relativity/relativity/Special%20Relativity_1_1.html

Just before the section 1.3.1. Michelson’s Interferometer

In my browser the caption for the “Fixed Ether” image is occluded by the TOC. Above that, I can see the TOC retracting and unrolling as I scroll. But not there, or just below that in any of the admonitions that I created called “Pencils Out!”

It also happens again after that with the overlapping sine wave images that follow...

thanks


On Dec 29, 2021, at 12:43 PM, Chris Holdgraf ***@***.******@***.***>> wrote:



Hmmm - in general, the right TOC should hide itself when there's margin content on the page. Do you have a link to a live site?

**Reply by choldgraf** [reply]
OK I think I've got a fix for this here: https://github.com/executablebooks/sphinx-book-theme/pull/448 once that's merged and released, give it another shot and see if this is improved?

**Reply by chipbrock** [reply]
Thanks. I will try it, but I upgraded to jupyter-books 0.12.1 and it broke margin figures, so I've got to figure out that. I'll separately post that issue.

**Reply by choldgraf** [reply]
try upgrading your sphinx-book-theme `pip install -U sphinx-book-theme` and re-build, and see if that fixes it (just made a release today)

**Reply by chipbrock** [reply]
Thanks for your attention. Actually, I think that did not fix it - the same spot that I linked to in the post  (and others) are still occluded as before. (Note: the link in the original post is now stale. I am working locally and have not sftp'd results to my server.) 

But it also seems to have broken linking from the right hand margin TOC contents for the chapter one is in. Clicking on them now does nothing...

**Reply by choldgraf** [reply]
Good catch - reported here: https://github.com/executablebooks/sphinx-book-theme/issues/451
